[SCRUM-191] The email history spreadsheet file name causes an email history error

sh1un opened the pull request:
## Description
<!--
請提供這個 PR 的簡要描述：
- 這個 PR 解決了什麼問題？
- 為什麼需要這個變更？
- 如何實現的？
-->

Fixed two issues:
1. Added null check handling for spreadsheet file name in email history list
2. Improved UI for better UX:
   - Enhanced recipient source display in email history list
   - Updated title and description in email details page
## Type of Change
<!--
請勾選適用的變更類型（可多選）
-->
- [x] 🐛 Bug Fix（修復 bug）
- [ ] ✨ New Feature（新功能）
- [x] 💄 UI/UX（介面或使用者體驗優化）
- [x] 🔨 Refactor（程式碼重構）
- [ ] 📝 Documentation（文件更新）
- [ ] 🔧 Configuration（配置變更）
- [ ] 🚀 Performance（效能優化）
- [ ] ✅ Test（測試相關）
- [ ] 🔒 Security（安全性相關）

## Related Issues
<!--
請列出相關的 issue 編號
例如：Closes #123, Relates to #456
-->

[SCRUM-191](https://aws-educate-cloud-ambassador-dev-team.atlassian.net/browse/SCRUM-191)

## Testing
<!--
請描述如何測試這些變更：
- 已執行哪些測試？
- 如何驗證這些變更？
- 是否需要特別的測試步驟？
-->
- You can go to the [preview environment](https://email-sender-hgdim0m9l-aws-educate-projects.vercel.app/emailHistory) to review the new UI


## Screenshots/Videos

1. emailHistory UI improvements
  - before
    <img width="1592" alt="image" src="https://github.com/user-attachments/assets/d96e1929-7caf-4b99-b842-b64a0d087f8a" />
  - after
    <img width="1573" alt="image" src="https://github.com/user-attachments/assets/281937a2-63bf-4423-8974-e71cad753fce" />
2. emailHistory/[runId] UI improvements
  - before
    <img width="485" alt="image" src="https://github.com/user-attachments/assets/f384a4df-370e-4037-876e-870d676d2cb7" />
  - after
    <img width="1596" alt="image" src="https://github.com/user-attachments/assets/b391a038-4a6a-4cb9-8d38-b974fb2cca5a" />



## Checklist
<!--
在提交 PR 前，請確認以下項目：
-->
- [x] 我已經測試過這些變更
- [ ] 我已經更新相關文件
- [x] 我的程式碼遵循專案的程式碼規範
- [x] 我已經處理所有的 console warnings/errors
- [ ] 我已經移除所有的 console.log
- [ ] 我已經新增必要的測試案例
- [x] 所有現有的測試都能通過

## Additional Notes
<!--
其他補充說明或注意事項
-->

N/A

[SCRUM-191]: https://aws-educate-cloud-ambassador-dev-team.atlassian.net/browse/SCRUM-191?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

chungchihhan commented on src/app/emailHistory/[runId]/page.tsx:
```diff
@@ -92,10 +91,10 @@ export default function Page({ params }: PageProps) {
         <div className="flex justify-between items-center w-full pb-4">
           <p className="text-gray-500 italic">
-            Details of <strong>one of the runs </strong>is shown here.
+            Detailed delivery status for run ID: <strong>{params.runId}</strong>
```
Is it neccessary to tell the user about the run ID?

chungchihhan commented on src/app/ui/email-history-card.tsx:
```diff
@@ -42,110 +18,163 @@ interface senderType {
                     <Send size={20} />
                     <p>Sender</p>
                   </div>
                   <strong>{item.sender.username}</strong>
```
Deal with the empty Sender 

<img width="1091" alt="image" src="https://github.com/user-attachments/assets/95f1bf49-7ccd-4916-a03c-398c2cd058f1" />
<img width="1090" alt="image" src="https://github.com/user-attachments/assets/8874da6a-8e97-4788-9701-008b460d5a88" />

chungchihhan commented on src/app/ui/email-history-card.tsx:
```diff
@@ -42,110 +18,163 @@ interface senderType {
 }
 
-export default function EmailHistoryCard({ data }: props) {
+export default function EmailHistoryCard({ data }: propsType) {
```
Better to return for empty or undefined data before `renderRecipientInfo`
```
  if (!data || data.length === 0) {
    return <div className="w-full p-8 text-center text-gray-500">No email history found</div>;
  }
```

chungchihhan commented:
The [preview](https://email-sender-hgdim0m9l-aws-educate-projects.vercel.app/emailHistory) here needs to login first. So try to navigate to the login page to login first. 

We got [SCRUM-193](https://aws-educate-cloud-ambassador-dev-team.atlassian.net/browse/SCRUM-193?atlOrigin=eyJpIjoiNjE4M2MyYzE2NzU5NGM3Mzg3Njg3MmJlMDg4MjYyMDIiLCJwIjoiaiJ9) to deal with this kind of login problem. So we will add this check in SCRUM-193

sh1un commented:
Hi @chungchihhan 

I've addressed all the feedback and pushed the changes. Please review agiain when you have time.


#### UI updates

1. Null sender ([comment link](https://github.com/aws-educate-tw/email-sender-nextjs/pull/50#discussion_r1899366555))
    <img width="1536" alt="image" src="https://github.com/user-attachments/assets/55b84c5e-381f-48df-8368-900ff422e3c2" />

2. Removed run ID ([comment link](https://github.com/aws-educate-tw/email-sender-nextjs/pull/50#discussion_r1899353684))
    <img width="442" alt="image" src="https://github.com/user-attachments/assets/23449d54-98c1-4b35-a04b-0f435aa7cbce" />

chungchihhan reviewed (approved):
# Code Review Checklist

## 基礎檢查
- [x] 確認自動化測試全部通過
- [x] 程式碼沒有合併衝突
- [x] PR描述清楚且完整

## 主要檢查項目
### 程式設計
- [x] 功能實作是否正確
- [x] 有適當的錯誤處理
- [x] 實作複雜度是否合理

### 程式品質
- [x] 變數/函數命名是否清晰
- [x] 程式碼容易理解
- [x] 符合專案程式風格

### 測試
- [x] 新功能有適當的測試
- [x] 已測試重要的邊界條件

### 文件
- [ ] 必要的文件是否更新
- [ ] API文件是否同步更新

## Review 建議事項：
N/A

## 整體評價
- [x] LGTM (Looks Good To Me) 
- [ ] 需要修改後再審查
